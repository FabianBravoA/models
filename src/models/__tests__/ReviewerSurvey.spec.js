const mongoose = require('mongoose');
const { ReviewerSurveySchema } = require('../ReviewerSurvey');
// const ReviewQuestionSchema = require('../ReviewQuestionSchema');


describe('ReviewerSurvey', () => {
  it('should ...', (done) => {
    const ReviewerSurveyModel = mongoose.model('ReviewerSurvey', ReviewerSurveySchema);
    // const ReviewQuestionModel = mongoose.model('ReviewQuestion', ReviewQuestionSchema);
    // const question = new ReviewQuestionModel({
    //   type: 'open',
    // });
    const reviewerSurvey = new ReviewerSurveyModel({
      // questions: [question],
    });
    reviewerSurvey.validate((err) => {
      expect(err).toBe(null);
      // console.log(reviewerSurvey);
      done();
    });
  });
});
