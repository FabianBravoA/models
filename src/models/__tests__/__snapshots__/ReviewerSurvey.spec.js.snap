// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewerSurvey should fail validation when multiple-choice missing options 1`] = `
Object {
  "questions": [Error: Options is required when type is multiple-choice],
}
`;

exports[`ReviewerSurvey should fail validation when question missing slug 1`] = `[ValidatorError: Path \`slug\` is required.]`;

exports[`ReviewerSurvey should fail validation when question options not a number 1`] = `
Object {
  "questions": [Error: Options is required when type is multiple-choice],
  "questions.0.options": [CastError: Cast to Number failed for value "hola" at path "options"],
}
`;

exports[`ReviewerSurvey should fail validation when wrong type 1`] = `
Object {
  "questions.0.type": [ValidatorError: \`other\` is not a valid enum value for path \`type\`.],
}
`;
